src/ServiceControl/Monitoring/EndpointMonitoringStats.cs

```diff
@@ -0,0 +1,16 @@
+namespace ServiceControl.Monitoring
+{
+    using System.Threading;
+
+    public class EndpointMonitoringStats
+    {
+        private int active;
+        private int failing;
+
+        public int Active => active;
+        public int Failing => failing;
+
+        public void RecordActive() => Interlocked.Increment(ref active);
```

Comment:
Actually my statement above is not quite true. `GetStats` can't be cached between updates. The endpoint status won't change but the monitored status might.